gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java

```diff
@@ -360,16 +361,23 @@ public Map<String, AddSpecResponse> put(Spec spec, boolean triggerListener) {
         responseMap.put(entry.getKey().getName(), entry.getValue().getResult());
       }
     }
+    AddSpecResponse<String> schedulerResponse = responseMap.getOrDefault(ServiceConfigKeys.GOBBLIN_SERVICE_JOB_SCHEDULER_LISTENER_CLASS, new AddSpecResponse<>(null));
```

Comment:
I've never looked closely at `AddSpecResponse`, but have grown quite weary in the course of this review.  those using it have been far more liberal with java raw types than I'm comfortable with (being basically never).

anyway, I see what you're doing to tunnel through the quota failure.  why not instead have the `GobblinServiceJobScheduler` (listener) throw the `QuotaFailureException` and you interrogate the `.getFailures()` here to see whether that happened?  if you find it, then rethrow (as a checked exception, which your resource handler needs to be on the look out for).  that way, no need to essentially disguise the situation and sneak it through; rather leave it explicit and therefore under the compiler's watch.

Reply:
in case helpful background/justification, consider the:
```
    return isCompileSuccessful(addSpecResponse.getValue());
```
in:
```
  public static boolean isCompileSuccessful(Map<String, AddSpecResponse> responseMap) {
    // If we cannot get the response from the scheduler, assume that the flow failed compilation
    AddSpecResponse<String> addSpecResponse = responseMap.getOrDefault(
        ServiceConfigKeys.GOBBLIN_SERVICE_JOB_SCHEDULER_LISTENER_CLASS, new AddSpecResponse<>(null));
    return isCompileSuccessful(addSpecResponse.getValue());
  }

  public static boolean isCompileSuccessful(String dag) {
    return dag != null && !dag.contains(ConfigException.class.getSimpleName());
  }
```
if `addSpecResponse` were not `AddSpecResponse<String>`, that invocation would give a runtime failure.  at the least it deserves a `instanceof`--yet raw types evade such enforcement.

primer, if helpful - https://stackoverflow.com/questions/2770321/what-is-a-raw-type-and-why-shouldnt-we-use-it